app/controllers/mappings_controller.rb

```diff
@@ -32,26 +31,27 @@ def create
   def update
     @mapping = Mapping.find(params[:id])
     authorize @mapping
+    # hack: set the user temporarily so that the model hook can reference it, and then set it back
+    temp = @mapping.user 
+    @mapping.user = current_user
+    @mapping.assign_attributes(mapping_params)
 
-    if @mapping.update_attributes(mapping_params)
+    if @mapping.save
       head :no_content
     else
       render json: @mapping.errors, status: :unprocessable_entity
     end
+    # restore the original mapping creator
+    @mapping.user = temp
+    @mapping.save
   end
 
   # DELETE /mappings/1.json
   def destroy
     @mapping = Mapping.find(params[:id])
     authorize @mapping
-
-    mappable = @mapping.mappable
-    if mappable.defer_to_map
-      mappable.permission = mappable.defer_to_map.permission
-      mappable.defer_to_map_id = nil
-      mappable.save
-    end
-
+    # hack: set the user temporarily so that the model hook can reference this user who is taking the action
+    @mapping.user = current_user
```

Comment:
They got moved into the before_destroyed model hook. They make it so that if the topic or synapse being removed from a map was deferring to it for its permissions, set the permission one last time but then disassociate them